STL_Extension/include/CGAL/iterator.h

```diff
@@ -37,28 +37,34 @@
 
 namespace CGAL {
 
-template<typename I>
+template<typename I, typename Value_type = I>
 class Prevent_deref
   : public boost::iterator_adaptor<
-  Prevent_deref<I>
+    Prevent_deref<I,Value_type>
   , I // base
-  , I // value
+  , Value_type // value
+  , boost::use_default
```

Comment:
The default here is `struct boost::iterators::detail::iterator_category_with_traversal<struct std::input_iterator_tag,struct boost::iterators::bidirectional_traversal_tag>`  and that triggers  a `static_assert`   `error C2338: prev requires bidirectional iterator`  [here ](https://github.com/CGAL/cgal/blob/94d45552bcace35592932e62c4196fbe79c2d4d0/Polygon_mesh_processing/test/Polygon_mesh_processing/test_pmp_clip.cpp#L475).

There is this issue on [github/boost](https://github.com/boostorg/iterator/issues/47)  explaining that this is expected behavior,  Shall we replace the `iterator_adaptor` by a hand written class?

Reply:
If the default is wrong, we can specify the category ourselves. But, would the resulting iterator be an actual forward iterator, with a real reference as reference?

Reply:
For `prev` all we want to know is if we can go in both directions I would say.

Reply:
[`std::prev`](https://en.cppreference.com/w/cpp/iterator/prev) requires a [_LegacyBidirectionalIterator_](https://en.cppreference.com/w/cpp/named_req/BidirectionalIterator), that refines [_LegacyForwardIterator_](https://en.cppreference.com/w/cpp/named_req/ForwardIterator), that requires that `reference` is an actual reference:

> Let `T` be the value type of `It`. The type `std::iterator_traits<It>::reference` must be either 
> 
>  - `T&` or `T&&` (since C++11) if `It` satisfies [_LegacyOutputIterator_](https://en.cppreference.com/w/cpp/named_req/OutputIterator) (`It` is mutable), or
>  - `const T&` or `const T&&` (since C++11) otherwise (`It` is constant), 
> 
>  (where T is the type denoted by `std::iterator_traits<It>::value_type`)

Reply:
I have fixed everything in 797c05657788608128be8d27012c4c0adda8f19f, from PR #7410.